### PR TITLE
Zombie Pig upgrades

### PR DESCRIPTION
**Pull request by Maleclypse**
#### Summary
Content "Zombie Pig Upgrades"

#### Purpose of change
Add's more variety to zombie types.  Specifically for pig forms.

#### Describe the solution
Skull Pigs are boar brutes and Trench pigs are the smoker variant.  New deathfunction for Trench Pigs.

#### Describe alternatives you've considered
Not finding ways to make the cataclysm resemble trench warfare.

### CODE REVIEW

File: data/json/monsters/zanimal_upgrade.json [hunk 1]
```diff
@@ -93,6 +93,43 @@
     ],
     "upgrades": { "half_life": 12, "into": "mon_skeleton_hulk" }
   },
+  {
+    "id": "mon_zombie_pig_brute",
+    "type": "MONSTER",
+    "name": { "str": "skull pig" },
+    "copy-from": "mon_zombie_pig",
+    "description": "This formerly wild boar appears to have been a truly massive specimen in life.  Stretching eight to nine feet in length, the most arresting feature of this animal is that it appears the horn of it's tusks have spread entirely across the face leaving a skull like appearance with deepset eyeholes.",
```
Comment (actual-nh) [inline data/json/monsters/zanimal_upgrade.json:101]:
```suggestion
    "description": "This former wild boar appears to have been a truly massive specimen in life.  Stretching eight to nine feet in length, the most arresting feature of this animal is that the enamel of its tusks has spread across its face leaving a skull-like appearance with deep-set eyeholes.",
```
Tusks are the canine teeth of the boar.

File: data/json/monsters/zanimal_upgrade.json [hunk 2]
```diff
@@ -93,6 +93,43 @@
...
+    "relative": {
+      "melee_dice": 1,
+      "melee_dice_sides": 5,
+      "melee_cut": 2,
+      "armor_bash": 4,
+      "armor_cut": 6,
+      "armor_bullet": 5,
+      "vision_night": 1
+    },
+    "upgrades": { "half_life": 21, "into": "mon_dog_zombie_hulk" },
```
Comment (actual-nh) [inline data/json/monsters/zanimal_upgrade.json:114]: Dog?

File: data/json/monsters/zanimal_upgrade.json [hunk 3]
```diff
@@ -93,6 +93,43 @@
...
+    "special_attacks": [ [ "SMASH", 30 ], { "type": "impale", "move_cost": 150, "cooldown": 15, "accuracy": 4, "no_infection_chance": 12 } ],
+    "extend": { "flags": [ "GROUP_BASH", "PUSH_VEH", "HIT_AND_RUN" ] }
+  },
+  {
+    "id": "mon_zombie_pig_gas",
+    "type": "MONSTER",
+    "name": { "str": "trench pig" },
+    "copy-from": "mon_zombie_pig",
+    "description": "Billowing clouds of yellow streaked gas precede boar shaped shadows.  Glimpses of a zombie boar are quickly obscured by the gases leaving it's body through open wounds.",
```
Comment (actual-nh) [inline data/json/monsters/zanimal_upgrade.json:123]:
```suggestion
    "description": "Billowing clouds of yellow-streaked gas precede boar-shaped shadows.  Glimpses of a zombie boar are quickly obscured by the gases leaving its body through open wounds.",
```

File: data/json/monsters/zanimal_upgrade.json [hunk 4]
```diff
@@ -93,6 +93,43 @@
...
+    "diff": 5,
+    "color": "red",
+    "harvest": "exempt",
+    "upgrades": { "half_life": 21, "into": "mon_dog_zombie_hulk" },
```
Comment (actual-nh) [inline data/json/monsters/zanimal_upgrade.json:127]: Again, dog?

**Review by actual-nh — COMMENTED** [review]
Interesting!

File: data/json/monsters/zanimal_upgrade.json (hunk 2 above)
Comment (Maleclypse) [reply data/json/monsters/zanimal_upgrade.json:114]: copy from error. thanks!

File: data/json/monsters/zanimal_upgrade.json [hunk 5]
```diff
@@ -93,6 +93,43 @@
...
+    "relative": {
+      "melee_dice": 1,
+      "melee_dice_sides": 5,
+      "melee_cut": 2,
+      "armor_bash": 4,
+      "armor_cut": 6,
+      "armor_bullet": 5,
+      "vision_night": 1
+    },
+    "upgrades": { "half_life": 21, "into": "mon_dog_zombie_hulk" },
```
Comment (Maleclypse) [inline data/json/monsters/zanimal_upgrade.json:114]:
```suggestion
   "upgrades": { "half_life": 20, "into_group": "GROUP_ZOMBIE_PIG_UPGRADE" },
```

File: data/json/monsters/zanimal_upgrade.json [hunk 6]
```diff
@@ -93,6 +93,43 @@
...
+  {
+    "id": "mon_zombie_pig_gas",
+    "type": "MONSTER",
+    "name": { "str": "trench pig" },
+    "copy-from": "mon_zombie_pig",
+    "description": "Billowing clouds of yellow-streaked gas precede boar-shaped shadows.  Glimpses of a zombie boar are quickly obscured by the gases leaving its body through open wounds.",
+    "diff": 5,
+    "color": "red",
+    "harvest": "exempt",
+    "upgrades": { "half_life": 21, "into": "mon_dog_zombie_hulk" },
```
Comment (Maleclypse) [inline data/json/monsters/zanimal_upgrade.json:127]:
```suggestion
    "upgrades": { "half_life": 20, "into_group": "GROUP_ZOMBIE_PIG_UPGRADE" },
```

**Comment by Maleclypse** [comment]
Note for self to address tomorrow.  :error file=data/json/monsters/zanimal_upgrade.json,line=115,col=62::unknown monster attack%0A%0A    },%0A    "upgrades": { "half_life": 20, "into_group": "GROUP_ZOMBIE_PIG_UPGRADE" },%0A    "special_attacks": [ [ "SMASH", 30 ], { "type": "impale",%0A                                                             ^%0A                                                              "move_cost": 150, "cooldown": 15, "accuracy": 4, "no_infection_chance": 12 } ],%0A    "extend": { "flags": [ "GROUP_BASH", "PUSH_VEH", "HIT_AND_RUN" ] }%0A  },%0A
(crafting_skill_gain)=>	05:54:01.798 ERROR : /home/runner/work/Cataclysm-DDA/Cataclysm-DDA/src/init.cpp:163 [void DynamicDataLoader::load_deferred(DynamicDataLoader::deferred_json&)] (json-error)
(crafting_skill_gain)=>	::error file=data/json/monsters/zanimal_upgrade.json,line=130,col=45::unknown monster attack%0A%0A    "emit_fields": [ { "emit_id": "emit_tear_gas_stream", "delay": "1 s" } ],%0A    "death_function": [ "TEARBURST" ],%0A    "special_attacks": [ { "type": "impale",%0A                                            ^%0A                                             "move_cost": 150, "cooldown": 20, "accuracy": 2, "no_infection_chance": 5 } ],%0A    "extend": { "flags": [ "HIT_AND_RUN" ] }%0A  },%0A

**Comment by Qrox** [comment]
Looks like `--error-format=github-action` does not work well with parallel execution of the unit tests.

**Comment by Maleclypse** [comment]
> Looks like `--error-format=github-action` does not work well with parallel execution of the unit tests.

Does that mean my error's aren't errors they are test errrors? Because at least when looking at them I'm having some trouble seeing what I'm doing wrong compared to things that are working.

**Comment by Qrox** [comment]
These errors are json errors. The format is supposed to tell github to print the error messages in the "files changed" tab, but the parallel execution adds prompt text before the error messages, so github is now unable to extract them.

**Pull request merged**